Frameworks/Foundation/NSSet.mm

```diff
@@ -279,15 +279,32 @@ - (instancetype)initWithArray:(NSArray*)array {
  @Status Interoperable
 */
 - (instancetype)initWithSet:(NSSet*)set {
+    return [self initWithSet:set copyItems:NO];
+}
+
+/**
+ @Status Interoperable
+*/
+- (instancetype)initWithSet:(NSSet*)set copyItems:(BOOL)flag {
+
     NSUInteger count = [set count];
     id* objects = (id*)IwMalloc(count * sizeof(id));
     NSUInteger i = 0;
 
-    for (id curObj in set) {
-        assert(i < count);
-        objects[i++] = curObj;
+    if(flag) {
+        for (id curObj in set) {
+            assert(i < count);
+            id obj = [curObj copy];
+            objects[i++] = obj;
+            [obj release];
```

Comment:
You can't release the object yet, or it might be deallocated before you insert it into the set.